Sparse Merkle Tree.md

```diff
@@ -2,49 +2,99 @@ This assumes prior knowledge of [Merkle Trees](https://en.wikipedia.org/wiki/Mer
 
-It is valuable to pre-compute a set of default hashes for all `q` levels:
+### Default hashes
+
+It is valuable to pre-compute a set of _default hashes_ for all `h` levels:
+
+- At level 0, the default hash is `H("")`
+- At level 1, the default hash is `H(H("") || H(""))`
+- At level 2, the default hash is `H(H(H("") || H("")) || H(H("") || H("")))`
+... and so on,
+where `H` is the hash function returning the result hash as an array of bytes, 
+`""` is 0-length array of bytes and `||` represents bytes concatenation.
 
-- At level 0, the default hash is `H(0)`
-- At level 1, the default hash is `H(H(0) || H(0))`
-- At level 2, the default hash is `H(H(H(0)|| H(0)) || H(H(0) || H(0)))`
-... and so on
+(TODO: Should we numerate the default hashes as D0, D1, etc?)
 
-In a sparse tree almost all level 0 nodes have the value `H(0)`, almost all level 1 nodes have the value `H(H(0) || H(0))`, almost all level 2 nodes have the value `H(H(H(0) || H(0)) || H(H(0) || H(0)))`, and so on. Only the nodes that lead to a non-zero leaf value actually need to be computed individually - the rest have the same value and we can compute them efficiently.
+In a sparse tree almost all leaves (level 0 nodes) have the value `H("")`, 
+almost all level 1 nodes have the value `H(H("") || H(""))`, 
+almost all level 2 nodes have the value `H(H(H("") || H("")) || H(H("") || H("")))`,
+and so on. Only the nodes that lead to a non-zero leaf value actually need to be 
+computed individually - the rest have the same value and we can compute them 
+efficiently.
 
-Here's an illustration of the simplified SMT storing 4-bit values. Here to store the keys `5` and `C` we only need to compute the hashes corresponding to white circles, all grey circles are default hashes and should not be duplicated in the storage.
-![Sparse Merkle Tree](assets/SMT.jpg)
+### Example 1
 
-## Sparse Merkle proofs
+Here's an illustration of the simplified SMT storing 4-bit values. 
+Here to store the keys `5` and `C` we only need to compute the hashes 
+corresponding to white circles, all grey circles are default hashes and 
+should not be duplicated in the storage.
+![Sparse Merkle Tree example image]
 
-Unlike a proof in a usual Merkle tree, proofs in SMT are NOT `q` N-bit hashes going all the way from level 0 at the leaf all the way to the root! 
+## Sparse Merkle Proofs
 
-Instead, a proof has a format `(proofBits, array_of_hashes)`, where  `proofBits`  is `q` bits that can compactly represent whether the sisters going up to the root are default hashes or not – that is, for each bit:
+Unlike a proof in an usual Merkle Tree, proofs in SMT are **not** `h` N-bit 
+hashes going all the way from level 0 at the leaf all the way to the root! 
 
-- 0 means "use the default hash"
-- 1 means "use the hash from `array_of_hashes`"
+Instead, a proof has a format `(proof_bits, array_of_hashes)`, 
+where  `proof_bits`  is `h` bits that can compactly represent whether the sisters (TODO: what?)
```

Comment:
Well I think we shouldn't explain the sisters here, as it should be clear from regular Merkle proofs.
But maybe ok to rephrase it somehow.